assets/src/edit-story/components/library/panes/text/textSets/textSets.js

```diff
@@ -66,13 +64,13 @@ const CATEGORIES = {
 function TextSets() {
   const [textSets, setTextSets] = useState([]);
   const [selectedCat, setSelectedCat] = useState(null);
+  const [filteredTextSets, setFilteredTextSets] = useState([]);
+  const [renderedTextSets, setRenderedTextSets] = useState([]);
```

Comment:
I needed both `filteredTextSets` and `renderedTextSets` to change states at roughly the same time so I converted `filteredTextSets` into a state variable (instead of `useMemo`) so they both can be updated within the same `useEffect`.

Reply:
Would using a `useReducer` for filtered and rendered buy us anything here?

Reply:
Good call, let me look, I think it would 🤔